CSC8503CoreClasses/PlayerObject.cpp

```diff
@@ -57,15 +57,18 @@ namespace {
 	constexpr bool DEBUG_MODE = true;
 }
 
-PlayerObject::PlayerObject(GameWorld* world, const std::string& objName,
-	InventoryBuffSystem::InventoryBuffSystemClass* inventoryBuffSystemClassPtr,
-	SuspicionSystem::SuspicionSystemClass* suspicionSystemClassPtr, PrisonDoor* prisonDoorPtr,
+PlayerObject::PlayerObject(GameWorld* world, InventoryBuffSystem::InventoryBuffSystemClass* inventoryBuffSystemClassPtr,
+	SuspicionSystem::SuspicionSystemClass* suspicionSystemClassPtr,
+	UISystem* UI,
```

Comment:
It's a good idea to pass the pointer into playerObject to avoid copying it.
But it would be better if it could be written as mUI.